src/source/Ice/Network.c

```diff
@@ -78,6 +78,12 @@ STATUS createSocket(PKvsIpAddress pHostIpAddress, PKvsIpAddress pPeerAddress, KV
     sockType = protocol == KVS_SOCKET_PROTOCOL_UDP ? SOCK_DGRAM : SOCK_STREAM;
 
     sockfd = socket(pHostIpAddress->family == KVS_IP_FAMILY_TYPE_IPV4 ? AF_INET : AF_INET6, sockType, 0);
+
+    if (sendBufLen > 0 && setsockopt(sockfd, SOL_SOCKET, SO_SNDBUF, &sendBufLen, SIZEOF(sendBufLen)) < 0) {
+        DLOGW("setsockopt() failed with errno %s", strerror(errno));
+        CHK(FALSE, STATUS_INTERNAL_ERROR);
```

Comment:
Why are we bailing out with an internal error? Doesn't seem right. This is a setsockopt failure which might still be OK. Also, I wonder if this can fail due to permissions?

Reply:
One more thing. Do we need to move the setsockopt after the check for failed sockfd below?